sdk/src/wallet/types/mod.rs

```diff
@@ -66,20 +66,18 @@ impl OutputData {
             .required_address(slot_index.into(), committable_age_range)?
             .ok_or(crate::client::Error::ExpirationDeadzone)?;
 
-        let chain = if &required_address == wallet_data.address.inner() {
-            self.chain
-        } else if required_address.is_ed25519() {
-            if wallet_data.address.inner() == &required_address {
-                // TODO #1279: do we need a check to make sure that `wallet_data.address` and `wallet_data.bip_path` are
-                // never conflicting?
+        // TODO #1279: do we need a check to make sure that `wallet_data.address` and `wallet_data.bip_path` are never
+        // conflicting?
+
+        let chain = self.chain.or_else(|| {
+            if &required_address == wallet_data.address.inner()
+                || required_address.backing_ed25519() == wallet_data.address.inner().as_ed25519_opt()
```

Comment:
edge case: what if both or `None`? :thinking:

Reply:
`wallet_data.address.inner().as_ed25519_opt()` can not be None at this point?

Reply:
chain implies ed25519, right? So can we remove the first part of the if?

Also to make extra sure we match `Some`s we could then just do:
`if matches!((required_address.backing_ed25519(), wallet_data.address.inner().as_ed25519_opt()), (Some(a), Some(b)) if Some(a) == Some(b)) { ... }`

Reply:
Honestly I don't like these `matches!` with an included `if`, I find them extremely hard to read

Reply:
you can't do that unfortunately :shrug: 

```rust
if let (Some(a), Some(b)) = (required_address.backing_ed25519(), wallet_data.address.inner().as_ed25519_opt()) && a == b { ... }
```

still ... the first part of the `if` isn't necessary, right?

Reply:
What first part? `required_address == wallet_data.address.inner()`? It's so we also check address that are not directly Ed25519